accepted/00000-minion-selective-action-blocking.md

```diff
@@ -0,0 +1,139 @@
+- Feature Name: minion_selective_action_blocking
+- Start Date: 2020-01-21
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+The goal is to selectively block certain action types from being executed on a Salt minion.
+
+# Motivation
+[motivation]: #motivation
+
+Main use cases:
+* Prevent certain action types when a product is installed on a minion, e.g. when a minion is a CaaSP node some actions shouldn't be allowed.
+* Feature parity with traditional clients. System locking is available for traditional clients but no such functionality exists for Salt minions.
+
+Usability:
+* Salt has a blackout mechanism. This is used for the first version of CaaSP support but it's not flexible enough for Uyuni's use case. Uyuni makes heavy use of `state.apply` to execute actions. Salt blackout allows whitelisting which modules are allowed during blackout. In the case of Uyuni, whitelisting `state.apply` means allowing basically all actions. Additionally the user experience of using Salt blackout is not great, the user can schedule any actions from the UI/API but then gets an error message at execution time that the action is not allowed by Salt.
+
+
+# Detailed design
+[design]: #detailed-design
+
+Traditional clients can be locked by the user. When locked, only certain action types are allowed. Actions that change the state of a system are blocked (package install, upgrade, apply highstate, etc) while read-only actions are allowed (hardware refresh, package profile refresh, etc).
+
+The same should be possible for Salt minions. However, in the case of Salt minions, additional flexibility is needed to allow blocking some actions depending on the the products installed on the minion. This is needed to accommodate CaaSP and maybe other clustering products in the future.
+
+To prevent the user from accidentally executing forbidden actions, the selective blocking must work both at the level of the Uyuni UI/XML-RPC API and at the level of the Salt command line.
+
+## Salt level selective action blocking
+
+The Salt blackout mechanism works by defining a special pillar called `minion_blackout` to tell the minion that it must reject all remote executions. Exceptions can be configured using the `minion_blackout_whitelist` pillar. This accepts a list of Salt functions that are allowed during blackout, e.g. `test.ping`, `state.apply`, etc.
+
+However, in Uyuni there are multiple actions that use `state.appply` under the hood. Some actions can change the state of the system while others are read-only but they all use `state.apply`. If `minion_blackout_whitelist` is used to allow `state.apply` then all actions that use this function are allowed.
+
+The blackout mechanism must be enhanced to allow a more fine grained filtering. One approach would be to check the metadata that can be attached to a Salt call. 
+
+A metadata whitelist would be defined. If the Salt call contains metadata that matches the whitelisted metadata then the call would be allowed. 
```

Comment:
@brejoc, is this big enough to warrant a Salt RFC from your perspective, or we just go ahead with a PR?